core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsZipFileInput.java

```diff
@@ -32,28 +33,43 @@
 public class GtfsZipFileInput extends GtfsInput {
   private final ImmutableSet<String> filenames;
   private final ZipFile zipFile;
+  private boolean isMacZip;
 
-  public GtfsZipFileInput(ZipFile zipFile) {
+  public GtfsZipFileInput(ZipFile zipFile, String zipFileName) {
     this.zipFile = zipFile;
-
+    StringBuilder strBuilder = new StringBuilder();
+    strBuilder.append(zipFileName);
+    strBuilder.append("/");
+    String macDirectory = strBuilder.toString();
     ImmutableSet.Builder<String> filenamesBuilder = new ImmutableSet.Builder<>();
     for (Enumeration<ZipArchiveEntry> entries = zipFile.getEntries(); entries.hasMoreElements(); ) {
       ZipArchiveEntry entry = entries.nextElement();
-      if (!isInsideZipDirectory(entry)) {
-        filenamesBuilder.add(entry.getName());
+      String entryName = entry.getName();
+      // check if the first entry is a directory with the name of the zip file
+      if (entry.isDirectory() && zipFileName.equals(entryName.replaceFirst("/", ""))) {
+        isMacZip = true;
+      }
+      Pair<String, Boolean> pair = isInsideZipDirectory(entry, macDirectory, entryName);
```

Comment:
I would not use a pair for that, just a boolean, and move this test inside the isInsideZipDirectory method.
I would also put a comment in the code as to why we test for ".DS_Store"

Reply:
Ok, make sense, I had this constraint of keeping isInsideZipDirectory when I coded the Pair stuff - it is removed now.